cli/flox-watchdog/src/logger.rs

```diff
@@ -15,26 +15,19 @@ const KEEP_WATCHDOG_DAYS: u64 = 3;
 const KEEP_SERVICES_LAST: usize = 5;
 
 /// Initializes a logger that persists logs to an optional file in addition to `stderr`
-pub(crate) fn init_logger(file_path: &Option<PathBuf>) -> Result<(), anyhow::Error> {
+pub(crate) fn init_logger(
+    logs_dir: &Option<PathBuf>,
+    log_file_prefix: &str,
+) -> Result<(), anyhow::Error> {
     let stderr_layer = tracing_subscriber::fmt::layer()
         .with_writer(std::io::stderr)
         .with_filter(EnvFilter::from_default_env());
-    let file_layer = if let Some(path) = file_path {
-        let path = if path.is_relative() {
-            std::env::current_dir()?.join(path)
-        } else {
-            path.clone()
-        };
-        let file = OpenOptions::new()
-            .create(true)
-            .truncate(true)
-            .write(true)
-            .open(&path)
-            .with_context(|| format!("failed to open log file {}", path.display()))?;
+    let file_layer = if let Some(dir_path) = logs_dir {
+        let appender = tracing_appender::rolling::daily(dir_path, log_file_prefix);
         Some(
             tracing_subscriber::fmt::layer()
                 .with_ansi(false)
-                .with_writer(file)
+                .with_writer(appender)
```

Comment:
question blocking: does this break the glob in `gc_logs_watchdog`?

Reply:
Almost certainly lol it would be cool if we had a test for gc'ing the logs